sockeye/inference.py

```diff
@@ -652,15 +670,19 @@ def make_input_from_multiple_strings(sentence_id: int, strings: List[str]) -> Tr
     """
     Returns a TranslatorInput object from multiple strings, where the first element corresponds to the surface tokens
     and the remaining elements to additional factors. All strings must parse into token sequences of the same length.
+
     :param sentence_id: An integer id.
     :param strings: A list of strings representing a factored input sequence.
     :return: A TranslatorInput.
     """
-    utils.check_condition(bool(strings), "List of strings cannot be empty.")
+    if not bool(strings):
+        return TranslatorInput(sentence_id=sentence_id, tokens=[], factors=None)
```

Comment:
TranslatorInput or _bad_input()?

Reply:
Update: This corresponds to the case where the input is indeed empty, right? Then it should be ok (is also handled with EMPTY_TRANSLATION below).

Reply:
I guess this is a case where the input is indeed empty so its fine as is.

Reply:
it was already a check_condition before, but I do agree, we should convert to the assert now that we're touching the code.

Reply:
done (but I think you were referring to the other comment below :))